### PR TITLE
add HTMLTemplateElement as an available polyfill

### PR DESCRIPTION
**Pull request by nicksrandall**
With the growing popularity of UI libraries like `lit-html` and `hyperHTML` which use HTMLTemplateElement under-the-hood, I think it makes sense to include a minimal polyfill for that here.

One potential concern I have is that the polyfill author does not ship a minified version of the library in their npm module. I assume that means that we will have to either minify it ourselves or serve the polyfill un-minified.

### CODE REVIEW

File: polyfills/HTMLTemplateElement/config.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+		"firefox": "4 - 22",
+		"opera": "10 - 22",
+		"safari": "6 - 9",
+		"chrome": "* - 35",
+		"ios_saf": "* - 9",
+		"firefox_mob": "<67",
+		"bb": "7 - 10",
+		"android": "<67"
+	},
+	"dependencies": ["~html5-elements", "DocumentFragment"],
```
Comment (JakeChampion) [inline polyfills/HTMLTemplateElement/config.json]: This polyfill does not depend on ~html5-elements as far as I can see.
This polyfill does depend on:
- DocumentFragment
- querySelectorAll
- querySelector
- Object.defineProperties
- Object.getOwnPropertyDescriptor
- Object.defineProperty
- document.addEventListener

File: polyfills/HTMLTemplateElement/polyfill.js [hunk 2]
```diff
@@ -0,0 +1,600 @@
+/**
```
Comment (JakeChampion) [inline polyfills/HTMLTemplateElement/polyfill.js]: We would need to add this filepath to .gitignore and .eslintignore

File: polyfills/HTMLTemplateElement/config.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
+	"browsers": {
+		"edge": "<12",
+		"edge_mob": "<12",
+		"ie": "*",
```
Comment (JakeChampion) [inline polyfills/HTMLTemplateElement/config.json]: Looking at the polyfill, I don't think it supports Internet Explorer 8 due to using getters, I may be wrong though.

**Comment by JakeChampion** [comment]
@nicksrandall Thank you for the pull-request, it is looking good so far.

Would it be possible for you to add some tests for the polyfill to ensure it works in all the browsers we have configured it to be served to? I'm really interested to see if this works in Internet Explorer 8, 9, and 10.

**Comment by nicksrandall** [comment]
@JakeChampion I added some tests but I only have the ability to test on IE 11 (everything worked). Are you able to run the tests on any older browsers? 

I also added the dependencies that you listed.

**Comment by nicksrandall** [comment]
I'm not sure what to do here... It looks like the polyfill only works for IE 11. Should I update the config.json to only include the polyfill for IE 11 since that is the only place that it is helpful?

**Comment by JakeChampion** [comment]
@nicksrandall Yep, I agree, I've updated the config to only serve IE 11 but can see that the tests fail in the Chrome versions that the polyfill is configured to serve, is that something you could look into at all?

**Comment by nicksrandall** [comment]
@JakeChampion I updated the config to be more specific on the versions where the polyfill works and is needed. We'll see how the next CI round goes.

**Comment by nicksrandall** [comment]
@JakeChampion I'm having a hard time reading these CI test results. I can't tell why it failed?

**Comment by JakeChampion** [comment]
Thanks for this contribution @nicksrandall!

**Comment by nicksrandall** [comment]
Would it be helpful if I created a PR to update the version of this library in polyfill-service?

**Pull request merged**